### PR TITLE
Add `canbus.packet` to `farm_ng_amiga`

### PR DESCRIPTION
**Pull request by Hackerman342**
Pulling `canbus.packet` out from #17 so it can be used by other repos

### CODE REVIEW

File: py/farm_ng/canbus/packet.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    STATE_ESTOPPED = 6
+
+
+def ticks_ms() -> int:
+    return int(time.monotonic() * 1000)
+
+
+def ticks_diff(a, b) -> int:
+    return int(a - b)
+
```
Comment (Hackerman342) [inline py/farm_ng/canbus/packet.py:31]: @ethanrublee 
Do you prefer to stamp these with a fake `ticks_ms` to be closer to circuitpy or just go for `time.monotonic()` stamps to be closer to the rest of the brain?

Comment (ethanrublee) [reply py/farm_ng/canbus/packet.py:31]: On linux, we should use the farm-ng core protobuf Timestamp struct.

Comment (ethanrublee) [reply py/farm_ng/canbus/packet.py:31]: and if you don't want this code to depend on protobuf for some reason, just a floating point which is the monotonic receive stamp.  But I would like to see this packet.py library use the protobuf message definitions directly for parsing.

Comment (Hackerman342) [reply py/farm_ng/canbus/packet.py:31]: Sounds great, I'll use: https://github.com/farm-ng/farm-ng-core/blob/main/py/farm_ng/core/stamp.py

**Comment by Hackerman342** [comment]
Request from @ethanrublee for free functions:
https://github.com/farm-ng/amiga-brain-api/pull/17#discussion_r1013157248

File: py/farm_ng/canbus/packet.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+from farm_ng.canbus import canbus_pb2
+from farm_ng.core.stamp import get_monotonic_now
+from farm_ng.core.timestamp_pb2 import Timestamp
+
+DASHBOARD_NODE_ID = 0xE
+PENDANT_NODE_ID = 0xF
+BRAIN_NODE_ID = 0x1F
+SDK_NODE_ID = 0x2A
+
+
```
Comment (edgarriba) [inline py/farm_ng/canbus/packet.py:17]: Why intenum?

File: py/farm_ng/canbus/packet.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+def make_amiga_rpdo1_proto(
+    state_req: AmigaControlState, cmd_speed: float, cmd_ang_rate: float
+) -> canbus_pb2.RawCanbusMessage:
+    """Creates a canbus_pb2.RawCanbusMessage, using the AmigaRpdo1 structure and formatting, that can be sent
+    directly to the canbus service to be formatted and send on the CAN bus."""
+    return canbus_pb2.RawCanbusMessage(
+        id=AmigaRpdo1.cob_id + DASHBOARD_NODE_ID,
+        data=AmigaRpdo1(state_req=state_req, cmd_speed=cmd_speed, cmd_ang_rate=cmd_ang_rate).encode(),
+    )
```
Comment (Hackerman342) [inline py/farm_ng/canbus/packet.py:62]: @ethanrublee I added util for creating a `canbus_pb2.RawCanbusMessage` (with `AmigaRpdo1` encoded) by providing state, speed, and angular rate

File: py/farm_ng/canbus/packet.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+def parse_amiga_tpdo1_proto(message: canbus_pb2.RawCanbusMessage) -> Optional[AmigaTpdo1]:
+    """Parses a canbus_pb2.RawCanbusMessage, IFF the message came from the dashboard and contains AmigaTpdo1
+    structure, formatting, and cobid.
+
+    Otherwise returns None.
+    """
+    if message.id != AmigaTpdo1.cob_id + DASHBOARD_NODE_ID:
+        return None
+    return AmigaTpdo1.from_can_data(message.data)
```
Comment (Hackerman342) [inline py/farm_ng/canbus/packet.py:141]: @ethanrublee and this util for returning `AmigaTpdo1` object from `canbus_pb2.RawCanbusMessage` iff the cobid matches

File: py/farm_ng/canbus/packet.py (hunk 2 above)
Comment (Hackerman342) [reply py/farm_ng/canbus/packet.py:17]: Sometime on the unpacking / decoding we get the corresponding int of the state value, not always the Enum. So this allows comparison. I feel like you'd see the same on protos sometimes...

**Comment by Hackerman342** [comment]
If approved, please merge into main

**Review by ethanrublee — APPROVED** [review]

**Pull request merged**